### ISSUE TITLE
Artiq starts with long delay on some boards

### ISSUE THREAD

**Issue by gkasprow** [sinara-hw/sinara#575]
@sbourdeauducq I noticed that on the board I got from @hartytp today the ARTIQ starts after several tens of seconds. Another board in my lab behaves in the same way. Other boards start immediately.  What enable the bootloader to start? What reset signal do you use?

**Comment by hartytp** [comment]
HMC830 wasn't locking

**Comment by marmeladapk** [comment]
No, it's before even Misoc logo appears in terminal.

**Comment by sbourdeauducq** [comment]
I would suspect there is some Sayma-bug around the flash that prevents the FPGA from reading it and loading its bitstream fast. Check the DONE pin to find out.
Note that if you tweaked the bitstream instead of building it straight from artiq/migen/misoc, you could have changed the flash readout settings too.

**Comment by gkasprow** [comment]
It starts itself after a minute. Then it works fine. I will have a look at
done pin

czw., 9 sie 2018, 17:58 użytkownik Sébastien Bourdeauducq <
notifications@github.com> napisał:

**Comment by gkasprow** [comment]
Btw,there is LED on the panel tied to the done pin

czw., 9 sie 2018, 18:14 użytkownik Grzegorz Kasprowicz <kasprowg@gmail.com>
napisał:

**Comment by jordens** [comment]
Are you saying there is a delay between DONE and the bootloader output?
There are only idelay calibrations and pll locks going on on that time.

**Comment by gkasprow** [comment]
No, I just said that there is LED tied to DON pin, but it means I have to check how much time it takes to configure.

**Comment by hartytp** [comment]
I never saw that on that board.

**Comment by gkasprow** [comment]
It looks like the configuration takes longer than usual. maybe FPGA reads entire memory instead of part of it?
 Eventually it manages to configure both FPGAs correctly.
On some boards I played with Vivado tools and programmed the FLASH memory, maybe Vivado left there something that now makes booting longer?

**Comment by jordens** [comment]
I don't see what could be left. If artiq has been flashed recently the relevant parts of the flash have been overwritten. But if you flash a artiq .bit using vivado then the loading options can be changed on the fly and can be slow.

**Comment by hartytp** [comment]
As I said, @gkasprow I've never seen this issue on that board using `artiq_flash` so I don't think there is a real hardware/artiq issue here.

**Comment by gkasprow** [comment]
No, I flashed it with artiq_flash recently, but before was playing with vivado tools.
So there is some issue which would be nice to explain before we produce v2.
